packages/backend-for-frontend/src/model/errors.ts

```diff
@@ -352,3 +358,62 @@ export function invalidZipStructure(description: string): ApiError<ErrorCodes> {
     title: "Invalid zip structure",
   });
 }
+
+export function missingActivePurposeVersion(
+  purposeId: string
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `There is no active version for purpose ${purposeId}`,
+    code: "missingActivePurposeVersion",
+    title: "Missing active purpose version",
+  });
+}
+
+export function activeAgreementByEserviceAndConsumerNotFound(
+  eserviceId: string,
+  consumerId: string
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Active agreement for Eservice ${eserviceId} and consumer ${consumerId} not found`,
+    code: "activeAgreementByEserviceAndConsumerNotFound",
+    title: "Active agreement not found",
+  });
+}
+
+export function multipleAgreementForEserviceAndConsumer(
+  eserviceId: string,
+  consumerId: string
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Multiple agreements for Eservice ${eserviceId} and consumer ${consumerId} found`,
+    code: "multipleAgreementForEserviceAndConsumer",
+    title: "Multiple agreement found",
+  });
+}
+
+export function purposeIdNotFoundInClientAssertion(): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `PurposeId not found in client assertion`,
```

Comment:
Could you log the ID of the purpose not found?

```suggestion
    detail: `Purpose ${purposeId} not found in client assertion`,
```

Reply:
I cannot log the purposeId since it was not found :) it is an assertion for getting the `purposeId` from the JWT token